src/test/java/bio/terra/workspace/service/workspace/flight/CreateGoogleContextFlightTest.java

```diff
@@ -0,0 +1,136 @@
+package bio.terra.workspace.service.workspace.flight;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import bio.terra.cloudres.google.cloudresourcemanager.CloudResourceManagerCow;
+import bio.terra.cloudres.google.serviceusage.ServiceUsageCow;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.FlightState;
+import bio.terra.stairway.FlightStatus;
+import bio.terra.stairway.Stairway;
+import bio.terra.workspace.common.BaseIntegrationTest;
+import bio.terra.workspace.common.StairwayTestUtils;
+import bio.terra.workspace.db.WorkspaceDao;
+import bio.terra.workspace.service.job.JobService;
+import bio.terra.workspace.service.workspace.WorkspaceCloudContext;
+import com.google.api.services.cloudresourcemanager.model.Project;
+import com.google.api.services.serviceusage.v1.model.GoogleApiServiceusageV1Service;
+import com.google.api.services.serviceusage.v1.model.ListServicesResponse;
+import java.time.Duration;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class CreateGoogleContextFlightTest extends BaseIntegrationTest {
```

Comment:
This test is blurring the line between unit and integration differently than the other tests in the repo.

This is using integration configs and creating real GCP projects (which feels like an integration test), but it's executing against the local service instead of a remote one, whereas all our integration tests use the client library. 

I'd rather make this a full integration test that runs against a remote server (though ack that it wouldn't be possible to write yet given that this functionality isn't exposed).

Reply:
I think there's a discussion to be had here. We need to be able to test cases that are we don't want to expose through an API. Case in point, the Error flight in this case - we don't really want an endpoint that creates flights that always fail. So, we need some notion of tests that don't run against a remote server. I'm open to creating new tests, or testing terminology - `locally connected` vs `remote`, but I think it's a mistake to try to make all integration tests run against remote servers.

Reply:
I see the testing need here, and I don't think we can cover everything with just unit tests and remote integration tests like we have now. My main objection is to blurring the test types together, which makes it hard to tell what services are being mocked vs. run locally vs. run remotely. 

I'd like to get other people's input on this, but one possibility is splitting this new test type (`connected`?) out into its own directory + profile + spring configuration + base testing class + gradle task, similar to the way we currently split integration tests out from unit tests. That's pretty heavyweight, but it would make the test types distinct.

Reply:
Per offline talk, going forward with a new test type `connected`. Going to circulate a doc soon describing these test types.
I've added a new profile/spring/base testing class/gradle task/github action.
I did not separate this into its own directory. I think the connected tests are often closely tied to a single class under test, as this one is for `CreateGoogleContextFlight`. Keeping it in the same package, similar to unit test, drives the connection home. That does mean that unit tests and connected tests will sometimes be in the same directory, but I think having an easy way to find the relevant test outweighs possible confusion.
PTAL.